cosmos-gui/src/oauth2.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2015 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-tidoop (FI-WARE project).
+ *
+ * fiware-tidoop is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ * fiware-tidoop is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with fiware-tidoop. If not, see
+ * http://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License please contact with
+ * francisco dot romerobueno at telefonica dot com
+ */
+
+/**
```

Comment:
Just out of my curiosity, which is the reason to use this implementation of OAuth instead of any of the packages already available in NPM (https://www.npmjs.com/search?q=oauth) There are even middlewares for Express implementing it if I am not wrong... :)